metals/src/main/scala/scala/meta/internal/metals/WorkDoneProgress.scala

```diff
@@ -0,0 +1,201 @@
+package scala.meta.internal.metals
+
+import java.util.UUID
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.MetalsEnrichments._
+
+import org.eclipse.lsp4j.ProgressParams
+import org.eclipse.lsp4j.WorkDoneProgressBegin
+import org.eclipse.lsp4j.WorkDoneProgressCreateParams
+import org.eclipse.lsp4j.WorkDoneProgressEnd
+import org.eclipse.lsp4j.WorkDoneProgressNotification
+import org.eclipse.lsp4j.WorkDoneProgressReport
+import org.eclipse.lsp4j.jsonrpc.messages
+import org.eclipse.lsp4j.services.LanguageClient
+
+class WorkDoneProgress(
+    client: LanguageClient,
+    time: Time,
+)(implicit ec: ExecutionContext)
+    extends Cancelable {
+  type Token = messages.Either[String, Integer]
```

Comment:
```suggestion
  type Token = messages.Either[String, Integer]
```
Its also defined in `object`